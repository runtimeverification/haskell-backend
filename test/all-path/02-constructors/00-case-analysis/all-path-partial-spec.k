requires "path.k"

module VERIFICATION
  imports PATH

endmodule

module ALL-PATH-PARTIAL-SPEC
  imports VERIFICATION
  imports PATH

  // This should not be provable both one-path and all-path.
  // because it does not hold for X = c
  claim <k> partial(X:S) => end </k> [all-path]

endmodule
