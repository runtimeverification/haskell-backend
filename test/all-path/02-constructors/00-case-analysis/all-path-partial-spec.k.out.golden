  #Not ( {
    X
  #Equals
    a
  } )
#And
  #Not ( {
    X
  #Equals
    b
  } )
#And
  <k>
    partial ( X ) ~> .K
  </k>
